apps/managedidentity/managedidentity_test.go

```diff
@@ -4,43 +4,244 @@ package managedidentity
 )
 
-func fakeClient(mangedIdentityId ID, options ...Option) (Client, error) {
-	client, err := New(mangedIdentityId, options...)
+func fakeMIClient(mangedIdentityId ID, options ...ClientOption) (Client, error) {
+	fakeClient, err := New(mangedIdentityId, options...)
 
 	if err != nil {
 		return Client{}, err
 	}
 
-	return client, nil
+	return fakeClient, nil
+}
+
+type errorClient struct{}
+
+func (*errorClient) CloseIdleConnections() {}
+func (*errorClient) Do(req *http.Request) (*http.Response, error) {
+	return nil, fmt.Errorf("expected no requests but received one for %s", req.URL.String())
+}
+
+type FakeClient struct {
+	responseType int
+}
+
+func (c *FakeClient) CloseIdleConnections() {}
+func (c *FakeClient) Do(req *http.Request) (*http.Response, error) {
+	println(c.responseType)
+	w := makeResponse(c.responseType)
+	return &w, nil
+}
+
+func makeResponse(responseType int) http.Response {
+	if responseType == 1 {
+		return http.Response{
+			StatusCode: http.StatusOK,
+			Body: io.NopCloser(strings.NewReader(`{
+      "access_token": "fakeToken",
+      "refresh_token": "",
+      "expires_in": "3599",
+      "expires_on": "1506484173",
+      "not_before": "1506480273",
+      "token_type": "Bearer"
+    }`)),
+			Header: make(http.Header),
+		}
+	} else {
+		return http.Response{
+			StatusCode: http.StatusBadRequest,
+			Body:       io.NopCloser(strings.NewReader(`{}`)),
+			Header:     make(http.Header),
+		}
+	}
 }
+func TestManagedIdentityIMDS_SAMISuccess(t *testing.T) {
+	fakeHTTPClient := FakeClient{responseType: 1}
 
-func TestManagedIdentity(t *testing.T) {
-	client, err := fakeClient(SystemAssigned())
+	client, err := fakeMIClient(SystemAssigned(), WithHTTPClient(&fakeHTTPClient))
 
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	_, err = client.AcquireToken(context.Background(), "scope", WithClaims("claim"))
+	result, err := client.AcquireToken(context.Background(), "fakeresource")
 
-	if err == nil {
-		t.Errorf("TestManagedIdentity: unexpected nil error from TestManagedIdentity")
+	if err != nil {
+		t.Fatal("TestManagedIdentity: unexpected nil error from TestManagedIdentity")
+	}
+
+	expected := accesstokens.TokenResponse{
+		AccessToken:  "fakeToken",
+		ExpiresOn:    internalTime.DurationTime{T: time.Now().Add(1 * time.Hour)},
+		ExtExpiresOn: internalTime.DurationTime{T: time.Now().Add(1 * time.Hour)},
+		TokenType:    "Bearer",
+	}
+	if result.AccessToken != expected.AccessToken {
+		t.Fatalf(`unexpected access token "%s"`, result.AccessToken)
 	}
+
 }
 
-func TestManagedIdentityWithClaims(t *testing.T) {
-	client, err := fakeClient(ClientID("123"))
+func TestCreateIMDSAuthRequest(t *testing.T) {
+	tests := []struct {
+		name     string
+		id       ID
+		resource string
+		claims   string
+		wantErr  bool
+	}{
+		{
+			name:     "System Assigned",
+			id:       SystemAssigned(),
+			resource: "https://management.azure.com",
+			claims:   "",
+			wantErr:  false,
```

Comment:
Just a minor style point: these assignments are unnecessary because the zero values of `string` and `bool` are `""` and `false`

Reply:
Thanks, updated this.
will take note of this for furture.